meta-lmp-base/recipes-core/initrdscripts/initramfs-framework/ostree_recovery

```diff
@@ -7,8 +7,44 @@ ostree_recovery_enabled() {
 }
 
 ostree_recovery_run() {
-	msg "Starting ostree recovery shell..."
-	sh
+	RECOVERY_MODULES_DIR=/recovery.d
+
+	# Load and run recovery modules same way as done for initramfs
+	for m in `ls $RECOVERY_MODULES_DIR/ | sort -n`; do
+		# Skip backup files
+		if [ "`echo $m | sed -e 's/\~$//'`" != "$m" ]; then
+			continue
+		fi
+
+		module=`basename $m | cut -d'-' -f 2`
+		debug "Loading recovery module $module"
+
+		# pre hooks
+		for h in $MODULE_PRE_HOOKS; do
+			debug "Calling recovery module hook (pre): $h"
+			eval "$h pre $module"
+			debug "Finished recovery module hook (pre): $h"
+		done
+
+		# process module
+		. $RECOVERY_MODULES_DIR/$m
+
+		if ! eval "${module}_enabled"; then
+			debug "Skipping recovery module $module"
+			continue
+		fi
+
+		debug "Running recovery ${module}_run"
```

Comment:
Yup, init is still the entry point as ostree_recovery is just another normal module for initramfs-framework.

Did that way in order to avoid having to create other modules such as udev, simply to get recovery going.

Reply:
Tested by adding shell in the boot command line arguments, and works the same way for initramfs and for recovery.

Reply:
got it, thanks for the explanation